url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/PropertyDescriptor.java

```diff
@@ -0,0 +1,300 @@
+                    && (pd.getReadMethod() == null) || (this.getter != null)
+                    && (this.getter.equals(pd.getReadMethod()));
+            boolean settersAreEqual = (this.setter == null)
+                    && (pd.getWriteMethod() == null) || (this.setter != null)
+                    && (this.setter.equals(pd.getWriteMethod()));
+            boolean propertyTypesAreEqual = this.getPropertyType() == pd
+                    .getPropertyType();
+            boolean propertyEditorClassesAreEqual = this
+                    .getPropertyEditorClass() == pd.getPropertyEditorClass();
+            boolean boundPropertyAreEqual = this.isBound() == pd.isBound();
+            boolean constrainedPropertyAreEqual = this.isConstrained() == pd
+                    .isConstrained();
+            result = gettersAreEqual && settersAreEqual
+                    && propertyTypesAreEqual && propertyEditorClassesAreEqual
+                    && boundPropertyAreEqual && constrainedPropertyAreEqual;
+        }
+        return result;
+    }
+
+    @Override
+    public int hashCode() {
+        return BeansUtils.getHashCode(getter) + BeansUtils.getHashCode(setter)
+                + BeansUtils.getHashCode(getPropertyType())
+                + BeansUtils.getHashCode(getPropertyEditorClass())
+                + BeansUtils.getHashCode(isBound())
+                + BeansUtils.getHashCode(isConstrained());
+    }
+
+    public void setPropertyEditorClass(Class<?> propertyEditorClass) {
+        this.propertyEditorClass = propertyEditorClass;
+    }
+
+    public Class<?> getPropertyType() {
+        Class<?> result = null;
+        if (getter != null) {
+            result = getter.getReturnType();
+        } else if (setter != null) {
+            Class<?>[] parameterTypes = setter.getParameterTypes();
+            result = parameterTypes[0];
+        }
+        return result;
+    }
+
+    public Class<?> getPropertyEditorClass() {
+        return propertyEditorClass;
+    }
+
+    public void setConstrained(boolean constrained) {
+        this.constrained = constrained;
+    }
+
+    public void setBound(boolean bound) {
+        this.bound = bound;
+    }
+
+    public boolean isConstrained() {
+        return constrained;
+    }
+
+    public boolean isBound() {
+        return bound;
+    }
+
+    String createDefaultMethodName(String propertyName, String prefix) {
+        String result = null;
+        if (propertyName != null) {
+            String bos = BeansUtils.toASCIIUpperCase(propertyName.substring(0, 1));
+            String eos = propertyName.substring(1, propertyName.length());
+            result = prefix + bos + eos;
+        }
+        return result;
+    }
+
+    void setReadMethod(Class<?> beanClass, String getterName)
+            throws IntrospectionException {
+        try {
+            Method readMethod = beanClass.getMethod(getterName, new Class[] {});
+            setReadMethod(readMethod);
+        } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)